pwndbg/heap/ptmalloc.py

```diff
@@ -32,6 +32,83 @@ def heap_for_ptr(ptr):
     return ptr & ~(HEAP_MAX_SIZE - 1)
 
 
+class Chunk:
+    def __init__(self, addr):
+        self._gdbValue = pwndbg.gdblib.memory.poi(pwndbg.heap.current.malloc_chunk, addr)
+        self.address = int(self._gdbValue.address)
+        self._prev_size = None
+        self._size = None
+        self._flags = None
+        self._fd = None
+        self._bk = None
+        # TODO fd_nextsize, bk_nextsize, key, REVEAL_PTR etc.
+
+    @property
+    def prev_size(self):
+        if self._prev_size is None:
+            if "mchunk_prev_size" in (field.name for field in self._gdbValue.type.fields()):
+                prev_size_field_name = "mchunk_prev_size"
+            else:
+                prev_size_field_name = "prev_size"
+
+            try:
+                self._prev_size = int(self._gdbValue[prev_size_field_name])
+            except gdb.MemoryError:
+                pass
+
+        return self._prev_size
+
+    @property
+    def size(self):
+        if self._size is None:
+            if "mchunk_size" in (field.name for field in self._gdbValue.type.fields()):
```

Comment:
Can we have a single function we call to handle all the renames? We could have a `dict` like (taken from elsewhere in the code):
```
        renames = {
            "mchunk_size": "size",
            "mchunk_prev_size": "prev_size",
        }
```

Then loop through the keys and replace with the other names if we don't find them.

Reply:
Good idea, added a function to handle this.
I've used a list within the dictionary to handle more than 2 names if any more renames occur in the future.